src/game/stage/intro.c

```diff
@@ -1,8 +1,1243 @@
+                DisplaySprite(s);
+            }
+        }
+    }
+}
+
+// (98.66%) https://decomp.me/scratch/gykfQ
+NONMATCH("asm/non_matching/game/stage/intro/Task_IntroActLettersAnimations.inc",
+         void Task_IntroActLettersAnimations(void))
+{
+    SITaskE *sit_e = TaskGetStructPtr(gCurTask);
+    u32 counter = sit_e->parent->counter;
+    Sprite *s;
+    u32 i;
+    s32 index;
+    s32 y;
+
+    if ((counter - 151) >= 40) {
+        if (counter >= 200) {
+            TaskDestroy(gCurTask);
+            return;
+        }
+    } else {
+        counter -= 150;
+
+        if (counter < 14) {
+            // _080302D8+6
+            for (i = 0; i < ARRAY_COUNT(sit_e->sprZoneNames); i++) {
+                s = &sit_e->sprZoneNames[i];
+
+                index = counter - i * 3;
+                if (index >= 4)
+                    index = 4;
+
+                index *= 8;
+
+                s->x = sScreenPositions_ZoneLoadingActLetters[i][0];
+
+                y = (index - 32);
+                s->y = y + sScreenPositions_ZoneLoadingActLetters[i][1];
+            }
+        } else if (counter < 18) {
+            // _0803031C+4
+            s32 shift;
+            counter -= 13;
+
+            shift = gUnknown_080D7130[counter];
+
+            for (i = 0; i < ARRAY_COUNT(sit_e->sprZoneNames); i++) {
+                s = &sit_e->sprZoneNames[i];
+                s->x = sScreenPositions_ZoneLoadingActLetters[i][0];
+                s->y = sScreenPositions_ZoneLoadingActLetters[i][1] + shift;
+            }
+        } else {
+            // _0803035C
+            for (i = 0; i < ARRAY_COUNT(sit_e->sprZoneNames); i++) {
+                s = &sit_e->sprZoneNames[i];
+                s->x = sScreenPositions_ZoneLoadingActLetters[i][0];
+                s->y = sScreenPositions_ZoneLoadingActLetters[i][1];
+            }
+        }
+        // _08030378
+        sub_8030488_inline();
+    }
+}
+END_NONMATCH
+
+static void TaskDestructor_StageIntroParent(struct Task *t)
+{
+    if (gCurrentLevel == LEVEL_INDEX(ZONE_1, ACT_1)) {
+        sub_8011328();
+    }
+
+    if (IS_SINGLE_PLAYER) {
+        gUnknown_03005424 &= ~EXTRA_STATE__DISABLE_PAUSE_MENU;
+    }
+}
+
+static void Task_UpdateStageLoadingScreen(void)
+{
+    SITaskB *sit_b = TaskGetStructPtr(gCurTask);
+    u32 counter = sit_b->parent->counter;
+
+    gBgPalette[0] = sZoneLoadingCharacterColors[gSelectedCharacter];
+
+    gFlags |= FLAGS_UPDATE_BACKGROUND_PALETTES;
+
+    if (counter >= 200) {
+        TaskDestroy(gCurTask);
+        return;
+    }
+}
+
+static void TaskDestructor_nop_8030458(struct Task *t) { }
+
+static void TaskDestructor_803045C(struct Task *t)
+{
+    SITaskD *sit_d = TaskGetStructPtr(t);
+    VramFree(sit_d->sprCharacterLogo.graphics.dest);
+}
+
+static void TaskDestructor_8030474(struct Task *t)
+{
+    SITaskE *sit_e = TaskGetStructPtr(t);
+    VramFree(sit_e->sprZoneNames[0].graphics.dest);
+}
+
+// NOTE: This only matches with the code being copy-pasted,
```

Comment:
I had this issue before. It would make sense that the inline function was defined here